### PR TITLE
migrated the backend lifecycle actor factory spec to batch

### PR DESCRIPTION
**Pull request by jarroyo28**
Migrated the BackendLifecycleActoryFactorySpec to batch. 
There are a few TODOs in the code to review whether some methods are required or not.

### CODE REVIEW

File: supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    val attributes = new GcpBatchConfigurationAttributes(
+      project = "project",
+      computeServiceAccount = "computeServiceAccount",
+      auths = null,
+      restrictMetadataAccess = true,
+      enableFuse = true,
+      executionBucket = "executionBucket",
+      // TODO: Determine if endpointUrl is necessary
+      endpointUrl = null,
```
Comment (dspeck1) [inline supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala]: endpoint url can be removed

File: supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      location = "location",
+      maxPollingInterval = 0,
+      qps = refineV[Positive](1).toOption.get,
+      cacheHitDuplicationStrategy = null,
+      requestWorkers = refineV[Positive](1).toOption.get,
+      // TODO: Determine if pipelineTimeout is necessary
+      pipelineTimeout = 1 second,
```
Comment (dspeck1) [inline supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala]: Update to batchTimeout

**Review by dspeck1 — CHANGES_REQUESTED** [review]
Looks good minor changes

File: supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala (hunk 1 above)
Comment (jarroyo28) [reply supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala]: Endpoint url has been removed

File: supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala (hunk 2 above)
Comment (jarroyo28) [reply supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala]: Pipeline timeout changed to batchtimeout

File: supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      (3, "exception message", () => throw new RuntimeException(actualRetryMessage)),
+      (1, "error not exception, no message", () => throw new Error()),
+      (1, "error not exception", () => throw new Error(actualRetryMessage))
+    )
+    forAll(fails) { (attempts, description, function) =>
+      it should s"$description: make $attempts attribute creation attempts before giving up" in {
+        val e = the [RuntimeException] thrownBy {
+//          GcpBatchBackendLifecycleActorFactory.robustBuildAttributes(function, initialIntervalMillis = 1, maxIntervalMillis = 5)
+        }
+        e.getMessage should startWith(s"Failed to build PipelinesApiConfigurationAttributes on attempt $attempts of 3")
```
Comment (dspeck1) [inline supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala]: Fix pipelines reference in text

**Review by dspeck1 — CHANGES_REQUESTED** [review]

File: supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala (hunk 3 above)
Comment (jarroyo28) [reply supportedBackends/google/pipelines/batch/src/test/scala/cromwell/backend/google/pipelines/batch/GcpBatchBackendLifecycleActorFactorySpec.scala]: Pipelines reference has been edited to 'GcpBatchConfigurationAttributes'

**Review by dspeck1 — APPROVED** [review]
looks good

**Pull request merged**